src/sagemaker/workflow/steps.py

```diff
@@ -228,7 +227,8 @@ class TrainingStep(ConfigurableRetryStep):
     def __init__(
         self,
         name: str,
-        estimator: EstimatorBase,
+        run_args: Dict = None,
```

Comment:
Some thoughts. Declaring this as Dict can be error prone for the users. There is no way for us to detect if users put a processing job args to a TrainingStep. I wonder if we can put a stronger type check to detect such errors at compile time. 

I'll spend some time to think about it.

Reply:
but this dict is generated from `.fit`, `.run`, `.transform` etc right? if the output from these methods is correct, it should not cause any error, unless customers manipulate this Dict directly.